File: services/vox-aegri/src/main/java/org/aegrotatio/types/enums/RequestLevelEnum.java
package org.aegrotatio.types.enums;

public enum RequestLevelEnum {
    LOW(1), LOW_TO_MEDIUM(2), MEDIUM(3), MEDIUM_TO_HIGH(4), HIGH(5);

    private final int level;

    RequestLevelEnum(int level) {
        this.level = level;
    }

    public int getLevel() {
        return level;
    }

    public static RequestLevelEnum fromLevel(int level) {
        for (RequestLevelEnum reqLevel : RequestLevelEnum.values()) {
            if (reqLevel.getLevel() == level) {
                return reqLevel;
            }
        }
        throw new IllegalArgumentException("Invalid request level: " + level);
    }
}

File: clients/dev-client/src/app/types/enums/request-level-enum.ts
export enum RequestLevelEnum {
  LOW = 1,
  LOW_TO_MEDIUM = 2,
  MEDIUM = 3,
  MEDIUM_TO_HIGH = 4,
  HIGH = 5
}

export const RequestLevelEnumNameMap: Record<RequestLevelEnum, string> = {
  [RequestLevelEnum.LOW]: 'LOW',
  [RequestLevelEnum.LOW_TO_MEDIUM]: 'LOW_TO_MEDIUM',
  [RequestLevelEnum.MEDIUM]: 'MEDIUM',
  [RequestLevelEnum.MEDIUM_TO_HIGH]: 'MEDIUM_TO_HIGH',
  [RequestLevelEnum.HIGH]: 'HIGH',
};
